Adding Azure Functions/Node create template

lostintangent opened the pull request:
## Why did you implement:

Closes #3333 (at least for Node.js)

## What did you implement:

I added a new template to the `create` plugin, which allows scaffolding a simple Node.js-based Azure Function, without needing to rely in `install` ing a Git repo.

This enables running the following command in order to get started quickly: `sls create -t azure-nodejs`

## How did you implement it:

Added a new template called `azure-nodejs` to the existing set of templates for the `create` plugin, and added a reference to this to the list of well-known templates.

I also updated the tests and docs to make use of the new `create` support.

## How can we verify it:

After setting the necessary [environment variables](https://serverless.com/framework/docs/providers/azure/guide/quickstart#2-set-up-credentials) for your Azure account, run the following commands, and verify that you see "Hello World" at the end:

```shell
$ sls create -t azure-nodejs --path <path>
$ cd <path>
$ npm i
$ sls deploy
$ sls invoke -f hello
"Hello World"
```

Alternatively, you could simply run `sls create -t azure-nodejs` and verify that it outputted valid `serverless.yml`, `handler.js`, and `package.json` files.

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

pmuens commented on lib/plugins/create/templates/azure-nodejs/serverless.yml:
```diff
@@ -0,0 +1,53 @@
+  hello: 
+    handler: handler.hello
+    events: 
+      - http: true
```
Can we showcase all the events here (commented out) so that this template is not `http` specific? Like it's done here: https://github.com/serverless/serverless/blob/d0b1195cffbcdcdc65e9fa6e69e3f99103fc2643/lib/plugins/create/templates/aws-nodejs/serverless.yml#L57-L83

lostintangent replied:
The `handler.js` code is fairly coupled to the event type though (even for AWS?), so it seemed to make sense to have the `serverless.yml` actually be functional from the start. Maybe `aws` defaults to `http` if an event isn't otherwise specified?

The AWS file you mentioned says that you'd need to change `handler.js` to adjust to any event changes. From that perspective, I was thinking it made sense to have the event definition be functional, and then have a comment that illustrates some other event types you can try. That makes the getting started/evaluation process super simple, and then you can tweak it based on your needs.

pmuens replied:
Thanks for the clarification. That makes sense!

pmuens commented on lib/plugins/create/templates/azure-nodejs/serverless.yml:
```diff
@@ -0,0 +1,53 @@
+        x-azure-settings:
+          authLevel : anonymous
+
+# The following are a few examples of other events you can configure:
```
We can then remove those entries here...

lostintangent commented:
@pmuens I addressed your documentation comments. I hadn't initially modified the integration tests, because I noticed the same hadn't been done for OpenWhisk, so I wasn't sure if that was a requirement. If you think this needs to be added as part of this PR, I'm happy to investigate it. Otherwise, I also added the basic unit test for the new template.

lostintangent commented:
@pmuens BTW, when I run the `test_all_templates` script as-is, I actually get the following error:

```
ERROR: Cannot start service serverless-node: oci runtime error: container_linux.go:247: starting container process caused "chdir to cwd (\"/app/tmp/serverless-integration-test-aws-csharp\") set in config.json failed: no such file or directory"
```

I believe the issue is with the working directory that is set by the `integration-test-template` when running Serverless commands (it seems like it should just be `/app`). Are these tests running fine on your end?

pmuens commented:
Thanks for the updates @lostintangent 👍 

I'll look into it ASAP!

lostintangent commented:
@pmuens What are your thoughts on merging this for now and investigating the integration test story in parallel? I added a unit test in this PR, and I'm happy to add an integration test as well, but they don't appear to be working for me right now, and that change would also involve your test environment to be setup with an Azure account as well.

lostintangent commented:
@pmuens Thanks so much!

pmuens commented:
Thanks for taking time and submitting a PR @lostintangent ! 💯
